upf/upf_cli.c

```diff
@@ -348,6 +348,9 @@ upf_nat_pool_add_del_command_fn (vlib_main_t * vm,
 				   port_block_size, min_port, max_port,
 				   is_add);
 
+      vec_free (nwi_name);
+      vec_free (name);
```

Comment:
We need to free these vectors in **any** case. Also, there's a similar problem with `nwi_s` which existed in the original code. One way to fix it: assign `0` to `name`, `nwi_name`, and `nwi_s` at the beginning of the func, assign `error` val and `goto done` instead of returning immediately upon an error, and at the `done` label, `vec_free()` them. Note that you do not need to check if the vars are `0` or not, as `vec_free()` does the check